lib/gollum-lib/filter/emoji.rb

```diff
@@ -0,0 +1,23 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Emoji
+#
+# Render emoji such as :smile:
+class Gollum::Filter::Emoji < Gollum::Filter
+  EMOJI_TAG_PATTERN = %r{
+    (?<!\[{2})
+    :(?:([\w-]|</?em>)+):
+    (?!\]{^2})
+  }ix
+
+  def extract(data)
+    data
+  end
+
+  def process(data)
+    data.gsub EMOJI_TAG_PATTERN do |tag|
```

Comment:
Could you use `gsub!` instead, or does the return value need to be non-nil? Same question for the second call to `gsub`.